src/components/buttons/Button.jsx

```diff
@@ -0,0 +1,159 @@
+   * @see type="destructive" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   * @see type="warning" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   * @see type="facebook" https://styleguide.brainly.com/latest/docs/interactive.html?type="link-button-inverted"#buttons
+   */
+  type: ButtonTypeType,
+  /**
+   * You can render icon inside each type of button on the left side
+   * @example <Button
+   *           icon={ICON_TYPE.FB}
+   *           buttonType={BUTTON_TYPE.FB}
+   *          >
+   *            Login with Facebook
+   *          </Button>
+   */
+  icon?: ?Node,
+  /**
+   * There are three sizes options for buttons, not need to be specify, default is medium
+   * @example <Button type="primary" size="medium">
+   *            button
+   *          </Button>
+   * @see size="small" https://styleguide.brainly.com/latest/docs/interactive.html?size="small"#buttons
+   * @see size="medium" https://styleguide.brainly.com/latest/docs/interactive.html?size="medium"#buttons
+   * @see size="large" https://styleguide.brainly.com/latest/docs/interactive.html?size="large"#buttons
+   */
+  size?: ButtonSizeType,
+  /**
+   * Specify href for button, optional string
+   * @example <Button href="https://brainly.com/" size="medium" type="primary-blue">
+   *            button
+   *          </Button>
+   */
+  href?: string,
+  /**
+   * Optional boolean for disabled button
+   * @example <Button type="primary-mint" disabled>
+   *            button
+   *          </Button>
+   */
+  disabled?: ?boolean,
+  /**
+   * Optional boolean for full width button
+   * @example <Button type="primary-mint" fullWidth>
+   *            button
+   *          </Button>
+   */
+  fullWidth?: ?boolean,
+  /**
+   * Additional class names
+   */
+  className?: string
+};
+
+const Button = ({
+  size,
+  type,
+  icon,
+  fullWidth,
+  disabled,
+  children,
+  className,
+  ...props
+}: ButtonPropsType) => {
+  const btnClass = classNames(
+    'sg-button', {
+      [`sg-button--${size || ''}`]: size,
+      [`sg-button--${type || ''}`]: type,
+      'sg-button--disabled': disabled,
+      'sg-button--full-width': fullWidth
+    },
+    className
+  );
+
+  const iconClass = classNames(
+    'sg-button__icon', {
+      [`sg-button__icon--${size || ''}`]: size
+    },
+    className
+  );
+
+  let ico;
+
+  if (icon) {
+    ico = <span className={iconClass}>{icon}</span>;
+  }
+
+  let TypeToRender = 'button';
+
+  if (props.href) {
+    TypeToRender = 'a';
+  }
+
+  return (
+    <TypeToRender
+      className={btnClass}
+      {...props}
+      disabled={disabled}
+      role={props.href ? 'button' : null}
+    >
+      {ico}
+      {/* As soon as we have Proxima fixed, we could remove that span */}
+      <span className="sg-button__text">{children}</span>
```

Comment:
To align text in the middle, we need to move the text down, as 'Proxima' font is broken. As soon (if ever) we get Proxima which is fixed (and there is one like that available), we could remove the span, which is kind of hack right now. Look into PR description.